src/flask/app.py

```diff
@@ -2318,6 +2320,19 @@ def test_request_context(self, *args, **kwargs):
         finally:
             builder.close()
 
+    def ensure_sync(self, func):
+        """Ensure that the returned function is sync and calls the async func.
+
+        .. versionadded:: 2?
+
+        Override if you wish to change how asynchronous functions are
+        run.
+        """
+        if asyncio.iscoroutinefunction(func):
```

Comment:
@pgjones Fix this with Python < 3.7

Reply:
This might be available in asgiref?

It seems like `asyncio.iscoroutinefunction` is only if you want to support the deprecated `@coroutine` decorated generator functions. Maybe use `inspect.iscoroutinefunction` since we only mention supporting `async def`?

Reply:
I've looked again and I don't think the fix is needed here, it is required in Quart due to the way I wrap functions.